ivc/src/ivc/columns.rs

```diff
@@ -0,0 +1,141 @@
+use crate::poseidon::columns::PoseidonColumn;
+use kimchi_msm::{
+    columns::{Column, ColumnIndexer},
+    fec::columns::FECColumn,
+    serialization::interpreter::N_LIMBS_SMALL,
+};
+
+pub const IVC_POSEIDON_STATE_SIZE: usize = 3;
+pub const IVC_POSEIDON_NB_FULL_ROUND: usize = 55;
+
+/// N := N_IVC + N_APP is the total number of columns in the circuit.
+///
+///```text
+///           34         34         34      8   8   8  4  4  4             constϕ
+///                                                  (repacked 150)          ϕ^i    same ϕs but
+///                                      (repacked 75)             (hashes)   r*ϕ^i  in 17 limbs       153
+///         Input1     Input2     Input3    R1  R2  R3 R1 R2 R3   H1 ... H6     -ϕ^i   each         FEC ADDs
+///  1   |----------|----------|----------|---|---|---|--|--|--|-------------|-|-|-|-|---|---|---|----------------|
+///      |   C_L1   |   C_L2   |   C_L3   |           |        |             |       |           |                |
+///      |   C_L4   |   C_L5   |   ...    |           |        |             |       |           |                |
+///      |   ...    |   ...    |   ...    |           |        |             |       |           |                |
+/// N/3  |----------|----------|----------|           |        |             |       |           |                |
+///      |   C_R1   |   C_R2   |   C_R3   |           |        |             |       |           |                |
+///      |   ...    |   ...    |   ...    |           |        |             |       |           |                |
+/// 2N/3 |----------|----------|----------|           |        |             |       |           |                |
+///      |   C_O1   |   C_O2   |   C_O3   |           |        |             |       |           |                |
+///      |   ...    |   ...    |   ...    |           |        |             |       |           |                |
+/// N    |----------|----------|----------|-----------|--------|-------------|-------|-----------|----------------|
+/// N+1  |        default_instance?       |
+///      |            ....                |              ...       FEC ADDs       ...
+///      |  ...??? empty space?
+///      |     reuse w/ selectors?
+/// 2^15 |---------------
+///```
+///
+/// Counting cells:
+/// - Inputs:              2 * 17 * 3N = 102N
+/// - Inputs repacked 75:  2 * 4 * 3N = 24N
+/// - Inputs repacked 150: 2 * 2 * 3N = 12N
+/// - Hashes:              2 * 165 * 3N = 990N (max 4 * 165 * 3N if we add 165 constants to every call)
```

Comment:
we also need to hash the challenges, but they are not that significant.

Reply:
Yep, you're right. I want to move somewhat iteratively. There are lots of things missing.